Sources/Penny/BotStateManager.swift

```diff
@@ -2,64 +2,123 @@ import DiscordBM
+        cancelIfCachePopulationTakesTooLong()
     }
-    
+
+    func cancelIfCachePopulationTakesTooLong() {
+        Task {
+            try await Task.sleep(for: .seconds(15))
+            if !isCachePopulated {
+                canRespond = true
+                logger.error("No CacheStorage-population signal was done in-time")
+            }
+        }
+    }
+
     func canRespond(to event: Gateway.Event) -> Bool {
-        checkIfItsASlowdownSignal(event: event)
+        checkIfItsASignal(event: event)
         return canRespond
     }
     
-    private func checkIfItsASlowdownSignal(event: Gateway.Event) {
+    private func checkIfItsASignal(event: Gateway.Event) {
         guard case let .messageCreate(message) = event.data,
               message.channel_id == Constants.Channels.logs.id,
               let author = message.author,
-              author.id.rawValue == Constants.botId,
-              message.content.hasPrefix(signal)
+              author.id.rawValue == Constants.botId
         else { return }
         guard let otherId = message.content.split(whereSeparator: \.isWhitespace).last else {
             logger.warning("Can't find id of the other Penny")
             return
         }
         if otherId == "\(self.id)" { return }
+        if message.content.hasPrefix(StateManagerSignal.shutdown.value) {
+            shutdown()
+        } else if message.content.hasPrefix(StateManagerSignal.didShutdown.value) {
+            populateCache()
+        } else {
+            logger.error("Unknown signal")
+            return
+        }
+    }
+
+    private func shutdown() {
         logger.warning("Received shutdown signal from another Penny")
         self.canRespond = false
         Task {
+            await cachesService.gatherCachedInfoAndSaveToRepository()
+            await send(.didShutdown)
+
             try await Task.sleep(for: disableDuration)
             self.canRespond = true
             logger.error("AWS has not yet shutdown this instance of Penny! Why?!")
         }
     }
-    
-    private func sendSignal() async {
+
+    private func populateCache() {
+        Task {
+            if isCachePopulated {
+                logger.warning("Received a did-shutdown signal but Cache is already populated")
+            } else {
+                isCachePopulated = true
+                canRespond = true
+                await cachesService.getCachedInfoFromRepositoryAndPopulateServices()
+            }
+        }
+    }
+
+    private func send(_ signal: StateManagerSignal) async {
         await DiscordService.shared.sendMessage(
             channelId: Constants.Channels.logs.id,
-            payload: .init(content: signal + " \(self.id)")
+            payload: .init(
+                content: makeSignalMessage(text: signal.value, id: self.id)
+            )
         )
     }
-    
+
+    func makeSignalMessage(text: String, id: Int) -> String {
+        "\(text) \(id)"
+    }
+
 #if DEBUG
     func _tests_reset() {
```

Comment:
Worth noting, generally having to do this means your unit tests aren't really "unit", since you're not testing actual code anymore.